corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -3,43 +3,66 @@ hqDefine('data_dictionary/js/data_dictionary.js', function () {
 
     var CaseType = function (name) {
         var self = this;
-        self.name = ko.observable(name);
+        self.name = name;
         self.properties = ko.observableArray();
 
         self.init = function (group_dict, changeSaveButton) {
             _.each(group_dict, function (properties, group) {
                 var groupObj = new PropertyListItem(group, true, group, self.name);
                 self.properties.push(groupObj);
                 _.each(properties, function (prop) {
-                    var propObj = new PropertyListItem(prop.name, false, prop.group, self.name, prop.data_type, prop.description);
+                    var propObj = new PropertyListItem(prop.name, false, prop.group, self.name, prop.data_type, prop.description, prop.deprecated);
                     propObj.description.subscribe(changeSaveButton);
                     propObj.dataType.subscribe(changeSaveButton);
+                    propObj.deprecated.subscribe(changeSaveButton);
                     self.properties.push(propObj);
                 });
             });
         };
     };
 
-    var PropertyListItem = function (name, isGroup, groupName, caseType, dataType, description) {
+    var PropertyListItem = function (name, isGroup, groupName, caseType, dataType, description, deprecated) {
         var self = this;
-        self.name = ko.observable(name);
+        self.name = name;
         self.expanded = ko.observable(true);
         self.isGroup = isGroup;
         self.group = ko.observable(groupName);
         self.caseType = caseType;
         self.dataType = ko.observable(dataType);
         self.description = ko.observable(description);
+        self.deprecated = ko.observable(deprecated || false);
         self.availableDataTypes = ko.observableArray([
```

Comment:
read the `initial_page_data` explanation [here](https://github.com/dimagi/js-guide/blob/master/integration-patterns.md) and offline with me if it doesn't make sense or you run into issues